service/controller/v22/resource/bridgezone/resource.go

```diff
@@ -176,13 +176,8 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "getting intermediate zone ID")
 
 		id, err := r.findHostedZoneID(ctx, defaultGuest, intermediateZone)
-		if IsNotFound(err) {
-			// If the intermeidate zone is not found we are after
-			// the migraiton period and this resource becomes noop.
+		if err != nil {
 			r.logger.LogCtx(ctx, "level", "debug", "message", "intermediate zone not found")
```

Comment:
I think this log should stay in `if IsNotFound(err) {` block. There may be unrelated errors.

Reply:
Then it gets redundant with the error handling, or we lose the information about the zone difference. Pick your evil.

Reply:
Just to be clear code below is what I meant. I don't think you lose any information with that.

```go
		if IsNotFound(err) {
			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource reconciliation for custom object")
			return microerror.Mask(err)
		} else if err != nil {
			return microerror.Mask(err)
		}
```

Reply:
The cancelation log was there before and it does not make sense to be there now IMO. Neither does the error checking. It is done below in one place. What is wrong with that? I can remove the log stating the zone is not found.

Reply:
We talked. Logging is improved.